### ISSUE TITLE
SSG Battlefront engine games

### ISSUE THREAD

**Issue by Python-Exoproject** [exoscoriae/eXoDOS#2029]
@SmilingSpectre & @exoscoriae 

There are 4 DOS games to this series:

Halls of Montezuma: A Battle History of the United States Marine Corps (1987 on Moby, 1988 game title screen, 1990 exodos label)
Rommel: Battles for North Africa (1988 on moby, 1989 exodos label)
MacArthur's War: Battles for Korea (1988)
Panzer Battles (1989 on moby, 1990 exodos label)

Two are capable of Tandy mode sound with machine=tandy and according to doc file adding "t" as command line option, though I didnt find that necessary. These games are essentially the same game with different included modules and title screens. I believe the other two should be capable and I feel the issues with running them in Tandy mode suspicious and its possible they are not complete copies, see notes below.

MacArthur's War: Battles for Korea: The other 3 games have music in Tandy and non Tandy modes, this one has none in any mode. Its missing the "Read(x).doc" files that all the others have, but still has a "readme.com". The title screen gets corrupted in Tandy mode which doesnt happen in the other 3.

Panzer Battles: While the sound outputs Tandy music when set to Tandy mode, the title screen half disappears, though the menu overlay still appears. If the game has Tandy music, it should have no problem with the titles screen.

Edit: One item I overlooked due to being distracted by all the other issues is that when things are working properly Tandy mode has a black background simple style game map while VGA\EGA has a full colour map. So that makes the VGA mode probably the best choice unless you want to add dual boot options. As the only needed change is machine=tandy you could do dual boot with just "-machine tandy" launch command.

This has revealed a problem with Halls of Montezuma: A Battle History of the United States Marine Corps that I hadnt noticed (and the main reason I didnt realise VGA was full colour), the VGA mode for this game isnt giving the full colour maps, just the same black background ones that Tandy mode gives. See here for what it should look like in VGA https://www.mobygames.com/game/halls-of-montezuma-a-battle-history-of-the-united-states-marine-/screenshots

**Comment by SmilingSpectre** [comment]
Our version of MacArthur is broken. TDC is less broken.

Unfortunately, it affects only two screens (title, and one of the maps). And I have no idea why it not works for Tandy.

Reference.

"Clear" CGA IFF (from Montezuma) looks like that:

![image](https://user-images.githubusercontent.com/24454178/94370883-42a78c00-00fb-11eb-9d59-aa2e482c8166.png)

MacArthur "clear" CGA is noisy (and Tandy uses exactly that):

![image](https://user-images.githubusercontent.com/24454178/94370905-610d8780-00fb-11eb-84bb-9324b0f3350e.png)

And our CGA is plain broken:

![image](https://user-images.githubusercontent.com/24454178/94370949-9619da00-00fb-11eb-9b98-0a24317b75dd.png)

**Comment by SmilingSpectre** [comment]
Hmm, but if I try to use Montezuma's file...

![image](https://user-images.githubusercontent.com/24454178/94371047-2fe18700-00fc-11eb-94ad-3945a60c0c57.png)

Something with .exe then. *shrug*

**Comment by SmilingSpectre** [comment]
Panzer Battles looks like a mess.  It have support for all 4 games' scenarios, but someone, seems, simply dumped all files from this games into it. So it have covers for all 4 games, and files are from everywhere (one readme is from MacArthur, second is from Rommel, for example).

Here is the "clean" install, but it's 3.0 (ours is 3.3). You can put all scenarios inside it from our installation (you need BTS+LBM pair), but i am not sure it will work for them. Need testing.

[Panzer_Battles.zip](https://github.com/exoscoriae/eXoDOS/files/5288746/Panzer_Battles.zip)

**Comment by Python-Exoproject** [comment]
That cleaner Panzer Battles is missing a chunk of the Scenarios that should have come with Panzer Battles, it only has the Eastern Theater ones. We could try and hack something out of both copies but at this point I think its all just one big mess with online copies and we should look for original copies if we can find them. In the meantime make the best of what weve got for the collection. If we ignore Tandy then the only real issues are the extra scenarios in Panzer Battles, the missing full colour gameplay in Montezuma,  and the missing sound in MacArthurs.

**Comment by Python-Exoproject** [comment]
I believe this pack https://www.amazon.com/Twenty-Wargame-Classics-DOS/dp/B0002XV6JE has what we want. Can be found with box here: https://www.ebay.com/itm/TWENTY-WARGAME-CLASSICS-inc-Panzer-General-1996-SSI-PC-CD-DOS-CIB-/283853318895 or just discs here: https://www.ebay.com/itm/Twenty-Wargame-Classica-SSI-PC-Games-1995-Windows/274306129069?hash=item3fddeb5cad:g:tf0AAOSw-e1ebEW9

**Comment by exoscoriae** [comment]
halls of montezuma is now tandy sound and date has been shifted to 1988.

**Comment by exoscoriae** [comment]
Rommel now dual boots Tandy and VGA

**Comment by exoscoriae** [comment]
I dumped a kryoflux image of the Halls of Montezuma disk. It isn't broken, it simply doesn't have full map color files on the disk. You'll notice it only has CGAPIC.IFF and EGAPIC.IFF. Rommel has a vga dat file in.

**Comment by exoscoriae** [comment]
I ordered the twenty war games classics discs. Putting this on ice until they arrive.

**Comment by exoscoriae** [comment]
damnit. ebay guy canceled the order because he lost the cd's.... 

i went and looked on RW though and found it there. downloading now.

**Comment by SmilingSpectre** [comment]
1. There _are_ 6 scenarios in the Panzer Battles. Proof:
![image](https://user-images.githubusercontent.com/24454178/94603234-a5855880-029e-11eb-9ba4-f92b2b971bcb.png)
2. But, it seems, our version have much more. Like, 30 more. Plus all scenarios from Rommel and MacArthur. Plus converted Montezuma scenarios (readme says originals are incompatible). Maybe it's the different version was issued later, but I cannot found anything about it. So I still suggest to use clean install and push all scenarios into it.
3. @exoscoriae , if you want, picture above taken from there: https://www.ebay.com/itm/Vintage-IBM-Tandy-Games-Rommel-Battles-For-North-Africa-Panzer-Battles-1989-1990/313170796627. Well, at least there will be 2 manuals, if nothing more.

**Comment by Python-Exoproject** [comment]
As @exoscoriae said this conversation has been a bit unclear I will try to summarise below:

--------------------------------------------------------------------------
Issue we have with the 4 Battlefront engine games are:

MacArthur's War: Battles for Korea

- No music in VGA or Tandy, 
- missing .doc files, 
- Tandy mode corrupts title screen.


Panzer Battles

- Tandy mode makes half of title screen disapear


Halls of Montezuma: A Battle History of the United States Marine Corps

- VGA mode missing full colour game maps


Rommel: Battles for North Africa

- Missing 4 scenarios: Alem al Halfa, Kasserine, Maknassy, Tebourga Gap (found here: [Missing Rommel Scenarios.zip](https://github.com/exoscoriae/eXoDOS/files/5305723/Missing.Rommel.Scenarios.zip)

-------------------------------------------------------------

Having seen your pic you posted I believe that the version of Panzer Battles found on "Twenty Wargame Classics" collection has the scenarios from all 4 games merged into it as an official release as the back of the box states "Use the award-winning land combat simulation system featuring: Panzer Battles, Rommel, Halls of Montezuma and MacArthur's War." But this means it wont be useful to us as we need the original game releases complete with their exe's to fix the above issues, not just their scenarios.

@SmilingSpectre Has uploaded a copy of Panzer Battles that fixes the Tandy mode issue and has just the 6 scenarios from the original Panzer Battles release. I think he then merged all of the current Panzer Battles scenarios from our copy and uploaded it. I have located the specific extra 20 scenarios that are only found in our collection (here: [Extra Scenarios from Panzer Battles.zip](https://github.com/exoscoriae/eXoDOS/files/5305810/Extra.Scenarios.from.Panzer.Battles.zip)) and suggest we have two copies of Panzer Battles, a clean one as originally released with the 6 scenarios, and another with just these 20 scenarios in it.

The above actions will just leave us with needing an original copy of MacArthur's War & Halls of Montezuma to fix those issues.

---------------------------------------------------------

That about sum it up @SmilingSpectre or have I misunderstood something you said?

**Comment by SmilingSpectre** [comment]
@Python-Exoproject 
1. If "Twenty Wargames" is the "correct" release (no trash, no mess, no half-screen modes), I believe it can be the best offer. Check it against combined version for the differences, maybe?
2. I am not sure if it worth to have two working versions differs only by having extras. Maybe have "vanilla" as extra archive? But even if do that - why 20 scenarios only? Why not have all available 50? (or 54? Are mentioned 4 Rommel scenarios missing?)

**Comment by exoscoriae** [comment]
MacArthur's War: Battles for Korea
- I ordered a clean copy from eBay. It will arrive around the 6th or 7th of Oct.

Panzer Battles
- I have no idea what to do here. Even the version in TDC has more files than the version @SmilingSpectre posted. It seems all these "extra" files quickly became part of the official release. 

Halls of Montezuma: A Battle History of the United States Marine Corps
- I've reached out to the mobygames submitter as he is active in the preservation community. I'll see what he says.

Rommel: Battles for North Africa
- I added the scenarios listed above

**Comment by Python-Exoproject** [comment]
Ive had a look at current synced versions and heres the missed items:

Hall of Montezuma: the Tandy works fine (only issue is the vga not being full colour), so tandy dual boot still needs setting up along with user note as per Rommel.

Panzer Battles: FYI I compared Twenty games version and its what we were running before. Ive noticed that current synced IBM version doesnt have full colour, not sure what happened but something went wrong between all the version discussions. I also noticed that both your 3.0 and 3.3 folders were actually running v3.0. Ive changed things as I think they should be and attached here: 
[PanzerBattles_Fixed.zip](https://github.com/exoscoriae/eXoDOS/files/5311876/PanzerBattles_Fixed.zip)
Hope that passes your QA and we can put this one to rest.

**Comment by Python-Exoproject** [comment]
Halls of Montezuma:

For a laugh I tried creating a version of the Twenty games release that includes just the Halls of Montezuma scenarios and it worked. I managed to get Montezuma title screens, scenarios, etc all running in VGA via the later releases files. The only catch is that it plays the intro song from the Twenty games release instead of the one from Montezuma. I kind of got around this by setting the Tandy version up with the actual Montezuma release and adding that the original song plays in Tandy mode to the user note. I know how you feel about hack jobs, but this is a borderline case in my mind as it is an official release, just cut down and focused on the scenarios that were in an earlier release. Up to you, here it is if you want to take a look: 
[HallsMontezuma_Test.zip](https://github.com/exoscoriae/eXoDOS/files/5312216/HallsMontezuma_Test.zip)

**Comment by exoscoriae** [comment]
ok. Panzer has been replaced with your version.
Your halls of montezuma seems fine. It's a hack job within the confines of original files. works for me if it works for you.

So now we are just waiting on the original MacArthur's War disks I believe.

**Comment by exoscoriae** [comment]
I got MacArthur in and all three disks read perfectly.

That said, the tandy screen is still broken.  This suggests the bug is endemic to the actual release of the game.

There is also no sound. lol.

I've added the disk images to the folder and the clean install to replace the old files if you want to take a closer look. I'm considering this wrapped up for now.

**Comment by Python-Exoproject** [comment]
Yeah, it looks like someone screwed the release for MacArthur. No point in setting up Tandy for it as the main advantage of the Tandy version was sound. Closed unless someone finds a working version, which is highly unlikely.

**Comment by Python-Exoproject** [comment]
I had a thought and tested it out, and it worked. Check out the attached file and see what you think as a solution. I approached it in a similar way to how I sorted Halls of Montezuma.
[MacArthurs_Test.zip](https://github.com/exoscoriae/eXoDOS/files/5355596/MacArthurs_Test.zip)

**Comment by exoscoriae** [comment]
fine with me. added.
